scribejava-httpclient-okhttp/src/main/java/com/github/scribejava/httpclient/okhttp/OAuthAsyncCompletionHandler.java

```diff
@@ -0,0 +1,102 @@
+package com.github.scribejava.httpclient.okhttp;
+
+import com.github.scribejava.core.model.OAuthAsyncRequestCallback;
+import com.github.scribejava.core.model.OAuthRequestAsync;
+import com.github.scribejava.core.model.Response;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Headers;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+class OAuthAsyncCompletionHandler<T> implements Callback, Future<T> {
+
+    private final OAuthAsyncRequestCallback<T> callback;
+    private final OAuthRequestAsync.ResponseConverter<T> converter;
+    private final Call call;
+    private final CountDownLatch latch;
+    private T result;
+
+    OAuthAsyncCompletionHandler(OAuthAsyncRequestCallback<T> callback,
+                                OAuthRequestAsync.ResponseConverter<T> converter, Call call) {
+        this.callback = callback;
+        this.converter = converter;
+        this.call = call;
+        this.latch = new CountDownLatch(1);
+
+        call.enqueue(this);
```

Comment:
it's better to create two classes. one to implement Callback and the second to implement Future.
So you can prevent from using 'this' in the constructor